src/core.rs

```diff
@@ -196,12 +188,51 @@ impl PartialEq for Kprobe {
 
 #[derive(Debug)]
 pub struct Tracepoint {
-    subsys: String,
-    name: String,
+    subsys: CString,
+    name: CString,
     code_fd: File,
     p: MutPointer,
 }
 
+impl Tracepoint {
+    fn attach(
```

Comment:
why `Tracepoint::attach` but `Kprobe::new`?

Reply:
I have a justification, but also did mangle that. I think this should actually be:
```rust
impl Tracepoint {
    pub fn attach_tracepoint(
```

The idea is that once we move `Kprobe`, `Uprobe`, and `Tracepoint` to their own modules, the `new()` constructor will be private. Instead, we'd use `Tracepoint::attach_tracepoint()` like we will `Kprobe::attach_kprobe()`. 

However, if you prefer, we can rename this to `Tracepoint::new()`, mark `Kprobe::new()` and `Uprobe::new()` as public, and just prefer the convenience functions of `Kprobe::attach_kprobe()`, `Kprobe::attach_kretprobe()`, and friends. I think we will leave `Kprobe`, `Uprobe`, and `Tracepoint` public only within the crate, and not re-export those publicly to users. Instead, users will still call the functions on the `BPF` struct. 

What are your thoughts on those options?

Reply:
renaming `attach` to `attach_tracepoint` sounds good to me!

Reply:
Awesome! Fixed in latest diff